phy/cluster/manual/session.py

```diff
@@ -76,59 +80,162 @@ def execute_action(self, action, *args, **kwargs):
 #------------------------------------------------------------------------------
 
 class FeatureMasks(StoreItem):
-    fields = [('features', 'disk'),
-              ('masks', 'disk'),
+    name = 'features and masks'
+    fields = [('features', 'disk', np.float32,),
+              ('masks', 'disk', np.float32,),
               ('mean_masks', 'memory'),
+              ('sum_masks', 'memory'),
               ('n_unmasked_channels', 'memory'),
+              ('main_channels', 'memory'),
               ('mean_probe_position', 'memory'),
               ]
-
-    def store_from_model(self, cluster, spikes):
-        # Only load the masks from the model if the masks aren't already
-        # stored.
-        to_store = {}
-
-        # Check if masks and features are already stored.
-        masks = self.store.load(cluster, 'masks')
-        features = self.store.load(cluster, 'features')
-        if (masks is None or masks.shape[0] != len(spikes) or
-                features is None or features.shape[0] != len(spikes)):
-            debug("Loading features and masks for "
-                  "cluster {0:d}...".format(cluster))
-            # Load features_masks from the KWX file.
-            n_features = self.model.metadata['nfeatures_per_channel']
-            n_channels = self.model.n_channels
-            fm = self.model.features_masks[spikes, ...]
-            features = fm[:, :, 0]
-            masks = fm[:, 0:n_features * n_channels:n_features, 1]
-            to_store.update(features=features, masks=masks)
-
-        # Extra fields.
-        mean_masks = masks.mean(axis=0)
-        n_unmasked_channels = (mean_masks > 1e-3).sum()
-        # Weighted mean of the channels, weighted by the mean masks.
-        mean_probe_position = (self.model.probe.positions *
-                               mean_masks[:, np.newaxis]).mean(axis=0)
-        to_store.update(mean_masks=mean_masks,
-                        n_unmasked_channels=n_unmasked_channels,
-                        mean_probe_position=mean_probe_position,
-                        )
-        self.store.store(cluster, **to_store)
-
-
-class Waveforms(StoreItem):
-    fields = [('spikes', 'memory'),
-              ('waveforms', 'custom')]
-
-    def store_from_model(self, cluster, spikes):
-        # Save the spikes in the cluster.
-        self.store.store(cluster, spikes=spikes)
-
-    def load(self, cluster, spikes=None):
-        if spikes is None:
-            spikes = self.store.load(cluster, 'spikes')
-            assert spikes is not None
-        return self.model.waveforms[spikes]
+    chunk_size = 100000
```

Comment:
what is this? why is this hard-coded here? perhaps a comment as to what this means and if people might want to change it